easybuild/easyblocks/s/score_p.py

```diff
@@ -38,49 +39,83 @@
+                toolchain.GCC: 'gcc',
+                toolchain.IBMCOMP: 'ibm',
+                toolchain.INTELCOMP: 'intel',
+                toolchain.PGI: 'pgi',
+            }
+            comp_fam = self.toolchain.comp_family()
+            if comp_fam in comp_opts:
+                self.cfg.update('configopts', "--with-nocross-compiler-suite=%s" % comp_opts[comp_fam])
             else:
-                raise EasyBuildError("MPI family %s not supported yet (only: %s)",
-                                     mpi_fam, ', '.join(mpi_opts.keys()))
+                raise EasyBuildError("Compiler family %s not supported yet (only: %s)",
+                                     comp_fam, ', '.join(comp_opts.keys()))
+
+            # --with-mpi=(bullxmpi|hp|ibmpoe|intel|intel2|intelpoe|lam|mpibull2|mpich|mpich2|mpich3|openmpi|
+            #             platform|scali|sgimpt|sun)
+            #
+            # Notes:
+            #   - intel:    Intel MPI v1.x (ancient & unsupported)
+            #   - intel2:   Intel MPI v2.x and higher
+            #   - intelpoe: IBM POE MPI for Intel platforms
+            #   - mpich:    MPICH v1.x (ancient & unsupported)
+            #   - mpich2:   MPICH2 v1.x
+            #   - mpich3:   MPICH v3.x & MVAPICH2
+            #               This setting actually only affects options passed to the MPI (Fortran) compiler wrappers.
+            #               And since MPICH v3.x-compatible options were already supported in MVAPICH2 v1.7, it is
+            #               safe to use 'mpich3' for all supported versions although MVAPICH2 is based on MPICH v3.x
+            #               only since v1.9b.
+            #
+            # With minimal toolchains, packages using this easyblock may be built with a non-MPI toolchain (e.g., OTF2).
+            # In this case, skip passing the '--with-mpi' option.
+            mpi_opts = {
+                toolchain.INTELMPI: 'intel2',
+                toolchain.OPENMPI: 'openmpi',
+                toolchain.MPICH: 'mpich3',     # In EB terms, MPICH means MPICH 3.x
+                toolchain.MPICH2: 'mpich2',
+                toolchain.MVAPICH2: 'mpich3',
+            }
+            mpi_fam = self.toolchain.mpi_family()
+            if mpi_fam is not None:
+                if mpi_fam in mpi_opts:
+                    self.cfg.update('configopts', "--with-mpi=%s" % mpi_opts[mpi_fam])
+                else:
+                    raise EasyBuildError("MPI family %s not supported yet (only: %s)",
+                                         mpi_fam, ', '.join(mpi_opts.keys()))
 
-        # auto-detection for dependencies mostly works fine, but hard specify paths anyway to have full control
+        # Auto-detection for dependencies mostly works fine, but hard specify paths anyway to have full control
         deps = {
-            'binutils': ['--with-libbfd=%%s/%s' % get_software_libdir('binutils', fs=['libbfd.a'])],
+            'binutils': ['--with-libbfd=%s'],
```

Comment:
@geimer: That sounds very reasonable, but... We're missing something.

On top of current `develop`, all builds using this `Score_minus_P` easyblock work fine; that includes: `Cube` (v4.x), `LWM2`, `OTF2`, `Scalasca` and `Score-P` itself.

On top of this updated easyblock however, two of the `Score-P` builds fail (`Score-P-1.2.1-gompi-1.4.12-no-OFED.eb` and `Score-P-1.2.3-goolf-1.5.14.eb`), while another one works fine `Score-P-1.4-foss-2015a.eb`... The other stuff using this easyblock works fine (`Cube` (v4.x), `LWM2`, `OTF2`, `Scalasca`).

So, it seems like we've introduced some kind of regression for Score-P 1.2.x?

Here's the relevant error message (reformatted for readability):

```
 == 2016-05-13 18:58:24,165 build_log.py:152 ERROR cmd " ./configure --prefix=...  --with-libbfd=$EBROOTBINUTILS " exited with exitcode 1 and output: 
...
checking bfd.h usability... yes                                                                                                                                                                                                                                           
checking bfd.h presence... yes                                                                                                                                                                                                                                            
checking for bfd.h... yes                                                                                                                                                                                                                                                 
: error: cannot find libbfd.a, libbfd.so, or libbfd.dylib                                                                                                                                                                                                                 
configure: error: ./configure failed for build-mpi  
```

Note that without this patch, it's not the absolute path of `libbfd.a` being passed, but the path to the directory in which `libbfd.a` resides, i.e. `--with-libbfd=$EBROOTBINUTILS/lib`.

This apparently works with Score-P 1.2.x and 1.4 (despite it being wrong according to the help info for `--with-libbfd` vs `--with-libbfd-lib`.